concordium-std/src/impls.rs

```diff
@@ -1606,3 +1623,200 @@ where
         source.read_u64().and_then(|set_prefix| Ok(StateSet::open(Rc::clone(state), set_prefix)))
     }
 }
+
+impl ContractStateApi for () {
```

Comment:
Given my comment on the `impl ContractStateApi for StateTrie`, I am not sure this makes so much sense to have.
The idea of wrapping the unsafe calls is fine, but if we do that, we should rewrite the other implementations (those not related to state) to also use this API. For example `slot_time()` in `ChainMetaExtern`. But changing those calls should not be part of the new state PR. (It's large enough already ;) )

Reply:
I don't think the idea makes sense. I do not see what value this abstraction would provide. Why is it valuable to provide a function with the same name that just wraps the unsafe? 

The functions are unsafe for a reason, and rely on correct usage, which is not ensured by this implementation. So these functions are in fact very unsafe and can lead to arbitrary memory corruption. For example `get_receive_sender` must have a buffer that is of correct size, which is not ensured here.